design-proposals/app-orch-deploy-applications.md

```diff
@@ -0,0 +1,209 @@
+
+  - `name`. Name/identifier for the `Registry`.
+  
+  - `kind`. Whether the registry is used for Helm Charts or Docker Images.
+
+  - `root_url`. The URL where charts are fetched from the registry.
+
+  - `username`, `password`, `cacerts`. For private registries, the credentails necessary to access
+    the registry.
+
+- `Application`. The `Application` object contains the following:
+
+  - `name`. Name/identifier for the `Application`
+
+  - `version`. Version number for the `Application`. Not necessarily the same as the version
+    number of the Helm Chart.
+
+  - `helm_registry_name`, `chart_name`, `chart_version`. These three fields together identify the
+    Helm Chart, by specifying a link to its `Registry` object, as well as the chart name and
+    version.
+  
+  - `image_registry_name`. An optional link to an image registry that contains username and
+    password for pulling docker images that are required by the helm chart.
+
+  - `Profiles`. A set of named yaml files that are used as `values.yaml` parameters when
+    deploying the helm chart. Each `Profile` supports `Parameter Templates`, which are an optional
+    feature to prompt the user to fill in values at deployment time. Multiple `Profiles` are
+    allowed, which allow the user to select between multiple canned configurations for the
+    `Application`.
+
+- `Deployment Package`. The `Deployment Package` object contains the following:
+
+  - `name`. Name/identifier for the `Deployment Package`.
+
+  - `version`. Version of the `Deployment Package`. Note that this is not necessarily the
+    same as the `Application` version, or the Helm Chart version.
+
+  - `applications`. A list of applications which comprise the `Deployment Package`. This is a
+    set of (`application_name`, `application_version`) tuples.
+
+  - `deployment_profiles`. A set of "master profiles", each of which contains a list of the
+    application profiles to be bundled together. For example, if there are three `Applications`
+    in the `Deployment Package`, then each `deployment_profile` would reference three
+    application profiles, one for each application.
+
+  - `is_deployed`. A boolean that keeps track of whether the `Deployment Package` is deployed.
+    Used to prevent deletion of assets that are in use.
+
+  - `default_namesspaces`. Used to pin applications to namespaces.
+
+  - `namespaces`. Used to create and label namespaces at deployment time.
+
+In Orchestrator 3.0, a single Helm Chart application requires all three objects to be created,
+usually in a 1:1 correspondence. The user has to create the `Registry`, then they create an
+`Application`, linking it to the `Registry`. Finally, the user creates the `Deployment Package`
+and links it the `Application`. There is complexity and overhead in creating the
+`deployment_profiles` as they have to shadow the application profiles.
+There is complexity in versioning all of these objects -- updating the
```

Comment:
I am not sure if its complex or cumbersome? or may be error prone.

Reply:
Complex, Cumbersome, _and_ error-prone.

Reply:
updated.